### PR TITLE
Run misspell linter with -w flag

### PR DESCRIPTION
**Pull request by cce**
## Summary

This runs the popular [misspell](https://github.com/client9/misspell) linter with the `-w` flag to automatically correct spelling mistakes in go-algorand.

## Test Plan

Only changes comments, should not impact test results.

### CODE REVIEW

File: test/scripts/e2e.sh [hunk 1]
```diff
@@ -50,10 +50,10 @@ echo "Test output can be found in ${TEMPDIR}"
 # Run kmd with scrypt() configured to run less secure and fast to go through the motions for test.
 # thus, on those platforms we launch kmd with unsafe_scrypt = true to speed up the tests.
 RUN_KMD_WITH_UNSAFE_SCRYPT=""
-PLATFORM_ARCHTYPE=$("${SRCROOT}/scripts/archtype.sh")
+PLATFORM_ARCHETYPE=$("${SRCROOT}/scripts/archtype.sh")
 
-echo "ARCHTYPE:    ${PLATFORM_ARCHTYPE}"
-if [[ "${PLATFORM_ARCHTYPE}" = arm* ]]; then
+echo "ARCHETYPE:    ${PLATFORM_ARCHETYPE}"
+if [[ "${PLATFORM_ARCHETYPE}" = arm* ]]; then
```
Comment (tsachiherman) [inline test/scripts/e2e.sh:56]: I don't think that you intended to have this change.

File: test/scripts/e2e_go_tests.sh [hunk 2]
```diff
@@ -83,9 +83,9 @@ cd ${SRCROOT}/test/e2e-go
 # ARM64 has some memory related issues with fork. Since we don't really care
 # about testing the forking capabilities, we're just run the tests one at a time.
 PARALLEL_FLAG=""
-ARCHTYPE=$("${SRCROOT}/scripts/archtype.sh")
-echo "ARCHTYPE:    ${ARCHTYPE}"
-if [[ "${ARCHTYPE}" = arm* ]]; then
+ARCHETYPE=$("${SRCROOT}/scripts/archtype.sh")
+echo "ARCHETYPE:    ${ARCHETYPE}"
+if [[ "${ARCHETYPE}" = arm* ]]; then
```
Comment (tsachiherman) [inline test/scripts/e2e_go_tests.sh:88]: same as above. I don't think you intended to have this change.

File: daemon/kmd/api/swagger.json [hunk 3]
```diff
@@ -644,7 +644,7 @@
       "x-go-package": "github.com/algorand/go-algorand/daemon/kmd/lib/kmdapi"
     },
     "APIV1POSTKeyExportResponse": {
-      "description": "APIV1POSTKeyExportResponse is the reponse to `POST /v1/key/export`\nfriendly:ExportKeyResponse",
+      "description": "APIV1POSTKeyExportResponse is the response to `POST /v1/key/export`\nfriendly:ExportKeyResponse",
```
Comment (tsachiherman) [inline daemon/kmd/api/swagger.json:647]: this is an auto-gen file, so you'll need to make the changes to the source files and re-run `make rebuild_swagger`

**Comment by tsachiherman** [comment]
I like the fixes.

File: test/scripts/e2e.sh (hunk 1 above)
Comment (cce) [reply test/scripts/e2e.sh:56]: thanks! I will get rid of it

**Review by algorandskiy — COMMENTED** [review]
Looks good, agree with Tsachi's remarks. Could you please see if it can be added as to Travis as part of "CodeGen Verification" build stage?

File: daemon/kmd/api/swagger.json (hunk 3 above)
Comment (cce) [reply daemon/kmd/api/swagger.json:647]: I see, the linter fixed it in both places but the rebuild triggered a rebuild of the bundled JSON spec, will commit and add it.

File: crypto/libsodium-fork/src/libsodium/crypto_aead/aes256gcm/aesni/aead_aes256gcm_aesni.c [hunk 4]
```diff
@@ -365,7 +365,7 @@ do { \
   */ \
     MAKE4(RED_MUL_MID); \
 \
-/* substracts x1*h1 and x0*h0 */ \
+/* subtracts x1*h1 and x0*h0 */ \
```
Comment (algorandskiy) [inline crypto/libsodium-fork/src/libsodium/crypto_aead/aes256gcm/aesni/aead_aes256gcm_aesni.c:368]: this is copy from libsodium, maybe we should not touch it at all :)

**Review by tsachiherman — APPROVED** [review]

**Review by algorandskiy — APPROVED** [review]
One comment about libsodium

File: crypto/libsodium-fork/src/libsodium/crypto_aead/aes256gcm/aesni/aead_aes256gcm_aesni.c (hunk 4 above)
Comment (tsachiherman) [reply crypto/libsodium-fork/src/libsodium/crypto_aead/aes256gcm/aesni/aead_aes256gcm_aesni.c:368]: I think that this is going to be deprecated along with the rest of the libsodium.. so I'm not particular picky around this one. But you're correct that it's not a "typo" that we've "generated" on our own. Maybe we should recommend libosdium to use this tool as part of their build toolchain.

**Pull request merged**